src/phoenix/datasets/types.py

```diff
@@ -165,48 +226,142 @@ def __post_init__(self) -> None:
             ValueError("Must specify either result or error")
 
 
-class _HasName(Protocol):
-    name: str
+ExampleOutput: TypeAlias = Mapping[str, JSONSerializable]
+ExampleMetadata: TypeAlias = Mapping[str, JSONSerializable]
+ExampleInputs: TypeAlias = Mapping[str, JSONSerializable]
```

Comment:
Why the plural here? I think all the other ones are of the same shape - seems okay to assume Input is a Mapping?